### PR TITLE
Has Games Filter, Minor build speed improvement

### PR DESCRIPTION
**Pull request by MaximumCrash**
Issue: #107 

This PR adds the "Has" filter feature for supporting if an entry contains..."something". 

For example, the Issue above requested having a filter for entries with games. The execution of this is to have an array of objects with fragments and a regex pattern based on the way the markdown data is organized. 

```
  const ExistsFilters = [{fragment: "Games", regex: /\[([^\[]+)\](\(.*\))/gm}]
```

We then map through the ExistsFilter array in a similar fashion to the filterFragments array, except this time we skip sanitizing the filter (santizing is only helpful for a specific context). 

With this execution we should be able to add more "Has" filters depending on the needs of our users. 

Screen Shot 2020-11-11 at 12 24 46 PM
Screen Shot 2020-11-11 at 12 24 35 PM

### CODE REVIEW

**Comment by cattsmall** [comment]
I like it!! I wonder if there's a copy change we can do so it's clear that the filter is for people who include games that can be played or purchased. Right now, I feel like "Has games" might not be super clear. "Games available" maybe? "Playable games"?

Otherwise LGTM

**Comment by MaximumCrash** [comment]
That's a good point, I'll tweak the copy and get this up to spec 👍

File: src/modules/layouts/SiteContext.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ const SiteProvider = ({ children, value }) => {
   `)
 
   const AllFilters = {}
+  const ExistsFilters = [{fragment: "Games", regex: /\[([^\[]+)\](\(.*\))/gm}]
```
Comment (annarankin) [inline src/modules/layouts/SiteContext.js]: Ahh this is testing for a markdown-formatted link like `[stuff](url)`? Nice! Would you mind adding a quick comment or extracting it into a `markdownLinkRegex` variable for clarity/people like me? 😂  the joys of regex

File: src/modules/layouts/SiteContext.js [hunk 2]
```diff
@@ -71,6 +72,38 @@ const SiteProvider = ({ children, value }) => {
       }
     })
 
+
+    //Map for "Has" filters.
+    //Checks if exists filters with their regex match 
+    //has a quantifiable amount to suggest an entry "Having [X]" 
+    ExistsFilters.map(({fragment, regex}) => {
+      if (node.rawBody.includes(`<${fragment}>`)) {
+        const matches = node.rawBody.match(regex); //Regex match based on fragment
```
Comment (annarankin) [inline src/modules/layouts/SiteContext.js]: If we're not going to use the capture groups and are just looking for a boolean, you could use `regex.test(node.rawBody)` I think and skip the length check below!

**Review by annarankin — APPROVED** [review]
Woot! Looks awesome, left two non-blocking/very smol suggestions!

**Pull request merged**